pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -53,6 +53,9 @@ type SpdyRoundTripper struct {
 
 	// Dialer is the dialer used to connect.  Used if non-nil.
 	Dialer *net.Dialer
+
+	// Proxier knows which proxy to use given a request
+	Proxier func(req *http.Request) (*url.URL, error)
```

Comment:
don't export this, comment that it is primarily for testing, and that the default uses `http.ProxyFromEnvironment`